core/src/main/java/tech/pegasys/eth2signer/core/multikey/DirectoryBackedArtifactSignerProvider.java

```diff
@@ -40,47 +47,79 @@
   private final Path configsDirectory;
   private final String fileExtension;
   private final SignerParser signerParser;
+  private final LoadingCache<String, ArtifactSigner> artifactSignerCache;
 
   public DirectoryBackedArtifactSignerProvider(
-      final Path rootDirectory, final String fileExtension, final SignerParser signerParser) {
+      final Path rootDirectory,
+      final String fileExtension,
+      final SignerParser signerParser,
+      final long maxSize) {
     this.configsDirectory = rootDirectory;
     this.fileExtension = fileExtension;
     this.signerParser = signerParser;
+    this.artifactSignerCache =
+        CacheBuilder.newBuilder()
+            .maximumSize(maxSize)
```

Comment:
what would be the behavior of -ve size? same as 0?

Reply:
negative values aren't allowed, Guava cache checks for this and will throw an IllegalArgumentException. So Eth2Signer will fail to startup in this case.

Reply:
👍